### PR TITLE
 Refactor gRPC path to use LoggingServiceV2Client.

### PR DESCRIPTION
**Pull request by qingling128**
PR #174 depends on this PR.

### CODE REVIEW

**Comment by qingling128** [comment]
This replaces https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/pull/172.

File: lib/fluent/plugin/out_google_cloud.rb [hunk 1]
```diff
@@ -1775,8 +1778,9 @@ def api_client
         authentication = Google::Auth.get_application_default
         creds = GRPC::Core::CallCredentials.new(authentication.updater_proc)
         creds = ssl_creds.compose(creds)
-        @client = Google::Logging::V2::LoggingServiceV2::Stub.new(
-          'logging.googleapis.com', creds)
+        channel = GRPC::Core::Channel.new('logging.googleapis.com', nil, creds)
+        @client = Google::Cloud::Logging::V2::LoggingServiceV2Client.new(
+          channel: channel)
```
Comment (igorpeshansky) [inline lib/fluent/plugin/out_google_cloud.rb]: [Optional] You can probably inline the `channel` expression in here.

File: test/plugin/test_out_google_cloud_grpc.rb [hunk 2]
```diff
@@ -257,114 +242,82 @@ def api_client
...
-  class GRPCLoggingMockService < Google::Logging::V2::LoggingServiceV2::Service
+  class GRPCLoggingMockService <
+      Google::Cloud::Logging::V2::LoggingServiceV2Client
     def initialize(requests_received)
       super()
       @requests_received = requests_received
     end
 
-    def write_log_entries(request, _call)
+    def write_log_entries(entries,
```
Comment (igorpeshansky) [inline test/plugin/test_out_google_cloud_grpc.rb]: This whole signature should fit on one line.

File: test/plugin/test_out_google_cloud_grpc.rb [hunk 3]
```diff
@@ -257,114 +242,82 @@ def api_client
...
     def initialize(code, message, failed_attempts)
       @code = code
       @message = message
       @failed_attempts = failed_attempts
       super()
     end
 
-    def write_log_entries(_request, _call)
+    # rubocop:disable Lint/UnusedMethodArgument
+    def write_log_entries(entries,
```
Comment (igorpeshansky) [inline test/plugin/test_out_google_cloud_grpc.rb]: This whole signature should fit on one line.

File: lib/fluent/plugin/out_google_cloud.rb (hunk 1 above)
Comment (qingling128) [reply lib/fluent/plugin/out_google_cloud.rb]: Changed.

File: test/plugin/test_out_google_cloud_grpc.rb (hunk 2 above)
Comment (qingling128) [reply test/plugin/test_out_google_cloud_grpc.rb]: Changed.

File: test/plugin/test_out_google_cloud_grpc.rb (hunk 3 above)
Comment (qingling128) [reply test/plugin/test_out_google_cloud_grpc.rb]: Changed.

**Comment by qingling128** [comment]
Just realized the last commit broke Rubocop. Fix is in.

**Comment by qingling128** [comment]
Friendly pin.

**Pull request merged**